python-threatexchange/threatexchange/signal_type/pdq/pdq_hasher.py

```diff
@@ -13,22 +13,23 @@
 ]  # hexadecimal representation of the Hash vector and a numerical quality value
 
 
-def pdq_from_file(path: pathlib.Path) -> PDQOutput:
+def pdq_from_file(path: pathlib.Path, force_rgb=False) -> PDQOutput:
```

Comment:
blocking: This needs a type (: bool) - thought we added a flag to the mypy settings to enforce this, but it looks like it didn't catch it. 

Additionally, in what circumstances will the caller know the correct option but that code won't be able to figure it out?

Reply:
Updated PR with typehints.

> Additionally, in what circumstances will the caller know the correct option but that code won't be able to figure it out?

I think our code should try to do the correct thing by default, but I think we need to confirm (have tests) around a few things:

1. How bad are the performance tradeoffs of always converting to RGB first? In our experience, eating the perf for converting to RGB isn't a problem. But I think there are performance benefits to resizing before conversion for example. I'm not innately familiar with pdq implementation so I'm not sure if pdq C bindings expect RGB.
2. Do we have test coverage for all "image types"? I'm hoping we (maybe I) can just add test coverage for all of these https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes There's also a difference between not raising exceptions (which currently this library is) and this library doing the right thing ie computing the correct hash given the visual appearance.